### PR TITLE
Relay server job data sending w/ cron job. 

### PR DESCRIPTION
**Pull request by tpolischuk**
Let's look at this together next week

### CODE REVIEW

File: blog-tutor-support.php [hunk 1]
```diff
@@ -88,12 +89,36 @@ protected function includes() {
...
+	}
+
 /**
  * Init the plugin.
  */
-add_action( 'plugins_loaded', array( 'NerdPress', 'get_instance' ) );
+	add_action( 'plugins_loaded', array( 'NerdPress', 'get_instance' ) );
+
+	// Register our custom action ping_relay so that it can be scheduled by cron
+	add_action( 'init', function () {
```
Comment (tpolischuk) [inline blog-tutor-support.php:109]: Move anonymous to named function (if possible)

File: includes/admin/class-support-admin.php [hunk 2]
```diff
@@ -183,6 +183,20 @@ public function settings_tabs() {
 			)
 		);
 
+		// Add option to schedule cron job to send snapshot
+		add_settings_field(
+			'schedule_snapshot',
```
Comment (tpolischuk) [inline includes/admin/class-support-admin.php:188]: Check if relay URL and relay key are present, disable button if so.

File: includes/admin-menu.php [hunk 3]
```diff
@@ -222,6 +222,26 @@ function bt_custom_toolbar_links( $wp_admin_bar ) {
...
 			);
 			$wp_admin_bar->add_node( $args );
+
+			if ( strpos( $_SERVER['REQUEST_URI'], '?' ) ) {
+				$query_char = '&';
+			} else {
+				$query_char = '?';
+			}
+
+			if ( isset( get_option( 'blog_tutor_support_settings' )['relay_key'] ) && isset( get_option( 'blog_tutor_support_settings' )['relay_url'] ) ) {
```
Comment (ecotechie) [inline includes/admin-menu.php:232]: Can we add this logic to a method in the NerdPress_Helpers Class? For better reusability.

File: includes/class-support-relay.php [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+		// If the request is a one-time call from the dashboard.
+		if ( isset( $_GET['ping'] ) ) {
+
+			$dump = self::assemble_dump();
+
+			$api_response = self::send_request_to_relay( $dump );
+
+			if ( $api_response['response']['code'] === 201 ) {
+				nocache_headers();
```
Comment (ecotechie) [inline includes/class-support-relay.php:52]: Move these two lines outside the if statement and log the error. So we can redirect (remove the ?ping) no matter the API result.

**Comment by tpolischuk** [comment]
I think this is ready to merge.

**Pull request closed without merging**